File: src/sets.py
se_algo = {"PHP", "Python", "SQL", "Shell", "QBasic", "Python"}
frameworks_web = {"Python", "PHP", "Java", "Elixir"}

# Intersección
puedo_usar = se_algo.intersection(frameworks_web)
print(puedo_usar)
# >>> {'Python', 'PHP'}

# Unión
todos_los_lenguajes = se_algo.union(frameworks_web)
# todos_los_lenguajes = se_algo | frameworks_web
print(todos_los_lenguajes)
# >>> {'Java', 'QBasic', 'SQL', 'PHP', 'Shell', 'JS', 'Python', 'Elixir'}

# Diferencia
diferencia = se_algo.difference(frameworks_web)
# >>> {'QBasic', 'SQL', 'JS', 'Shell'}

File: src/sets.php
<?php

// En PHP no hay sets, entonces tenemos que asegurarnos de que
// los elementos en el array sean únicos con: array_unique()
$se_algo = ["PHP", "Python", "JS", "SQL", "Shell", "QBasic"];
$frameworks_web = ["Python", "PHP", "Java", "Elixir"];

// Intersección
$puedo_usar = array_intersect($se_algo, $frameworks_web);
var_dump($puedo_usar);
// "PHP", "Python"

// Unión
$todos_los_lenguajes = array_unique(array_merge($se_algo, $frameworks_web));
var_dump($todos_los_lenguajes);
// "PHP", "Python", "JS", "SQL", "Shell", "QBasic", "Java", "Elixir"

// Diferencia
$diferencia = array_diff($se_algo, $frameworks_web);
var_dump($diferencia);
// "JS", "SQL", "Shell", "QBasic"


